kube-runtime/src/events.rs

```diff
@@ -64,6 +74,34 @@ pub enum EventType {
     Warning,
 }
 
+/// ObjectReference with Hash and Eq implementations
+#[derive(Clone, Debug, PartialEq)]
+pub struct Reference(ObjectReference);
+
+impl Eq for Reference {}
+
+impl Hash for Reference {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.0.api_version.hash(state);
+        self.0.kind.hash(state);
+        self.0.name.hash(state);
+        self.0.namespace.hash(state);
+        self.0.uid.hash(state);
+    }
+}
+
+/// Isomorphic key for caching similar events
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct EventKey {
```

Comment:
does this need to be pub?

Reply:
Probably not. Fixed in bebbb103967e201d7a393ae1865f4e21bcd72c42